starlight-rabbitmq/src/main/java/com/datastax/oss/starlight/rabbitmq/GatewayService.java

```diff
@@ -487,4 +490,27 @@ public ScheduledExecutorService getExecutor() {
   public ExecutorService getInternalExecutorService() {
     return internalExecutorService.getExecutor();
   }
+
+  public Producer<byte[]> getOrCreateProducer(String topic) {
+    return producers.computeIfAbsent(topic, this::createProducer);
```

Comment:
Should this fail if the gateway service is closing ?

Reply:
These is no need of we are shutting down the client